### PR TITLE
ARM Mac open-source and packaged build

### PR DESCRIPTION
**Pull request by jonsimantov**
This enables cross-compiling for arm64 Mac. This changes the default Xcode version to 12.2, as Xcode 11 doesn't support arm64, and 12.0 fails with vcpkg. It'll still build with 11.7 in the expanded matrix and omit arm64.

Unit tests are built, but not run, on arm64. (They can't be run until GitHub has hosted ARM Mac runners, or we add our own.)

This PR adds a --arch option to the testapp builder but does not yet build for arm64 in the integration_tests workflow.

For packaged builds, we now build LLVM binutils on Mac, in addition to GNU binutils. LLVM binutils are required for the merge_libraries step on Mac, as GNU binutils does not support outputting ARM64 Mach-O files properly. (We still need GNU binutils on Mac for c++filt). Because LLVM binutils can take a long time to build, we cache it.

### CODE REVIEW

**Comment by jonsimantov** [comment]
Successful packaging run: https://github.com/firebase/firebase-cpp-sdk/actions/runs/1268032776

Integration test failures in both source and packaged runs are flakes only.

File: .github/workflows/cpp-packaging.yml [hunk 1]
```diff
@@ -105,13 +107,47 @@ jobs:
...
+
+      - name: Build LLVM (Mac only) 
+        # Only build LLVM if we don't already have binaries from the cache.
+        if: ${{ matrix.tools_platform == 'darwin' && !steps.cache_llvm.outputs.cache-hit }}
+        run: |
+          mkdir llvm-src/llvm/build
+          cd llvm-src/llvm/build
+          cmake -DLLVM_ENABLE_PROJECTS=clang -DCMAKE_BUILD_TYPE=Release ..
+          cmake --build . -j 3 --target llvm-objcopy --target llvm-nm --target llvm-ar
+          cd -
```
Comment (DellaBitta) [inline .github/workflows/cpp-packaging.yml:144]: is this `cd` needed?

File: release_build_files/readme.md [hunk 2]
```diff
@@ -337,40 +337,41 @@ information).
...
-|                               | firebase.xcframework
-Firebase Dynamic Links (stub)   | firebase_dynamic_links.xcframework
-|                               | firebase.xcframework
-Firebase Installations (stub)   | firebase_installations.xcframework
-|                               | firebase.xcframework
-Firebase Cloud Messaging (stub) | firebase_messaging.xcframework
-|                               | firebase.xcframework
-
-The provided libraries have been tested using Xcode 12.0. When building C++
+Firebase Authentication         | firebase_auth.framework
```
Comment (DellaBitta) [inline release_build_files/readme.md:347]: We're not providing xcframeworks anymore?  I thought they were the new thing, not the old thing..?

**Review by DellaBitta — APPROVED** [review]
Approved with some questions.

File: release_build_files/readme.md (hunk 2 above)
Comment (jonsimantov) [reply release_build_files/readme.md:347]: We are for iOS. We aren't for OSX, and never were. This was in error.

File: .github/workflows/cpp-packaging.yml (hunk 1 above)
Comment (jonsimantov) [reply .github/workflows/cpp-packaging.yml:144]: Ah I guess not, it's leftover from before the caching when it was 1 step

File: release_build_files/readme.md (hunk 2 above)
Comment (DellaBitta) [reply release_build_files/readme.md:347]: Ahhh, I see.

**Comment by jonsimantov** [comment]
This enables cross-compiling for arm64 Mac. This changes the default Xcode version to 12.2, as Xcode 11 doesn't support arm64, and 12.0 fails with vcpkg. It'll still build with 11.7 in the expanded matrix and omit arm64.

Unit tests are built, but not run, on arm64. (They can't be run until GitHub has hosted ARM Mac runners, or we add our own.)

This PR adds a --arch option to the testapp builder but does not yet build for arm64 in the integration_tests workflow.

For packaged builds, we now build LLVM binutils on Mac, in addition to GNU binutils. LLVM binutils are required for the merge_libraries step on Mac, as GNU binutils does not support outputting ARM64 Mach-O files properly. (We still need GNU binutils on Mac for c++filt). Because LLVM binutils can take a long time to build, we cache it.

**Pull request merged**